### DISCUSSION TITLE
Adding Speaker Notes to Title Slide of Revealjs Presentation

### DISCUSSION THREAD

**Discussion by DeanVogt** [quarto-dev/quarto-cli#4824] [Q&A] [answered]
Is it possible to add speaker notes to the **title** slide of a Revealjs presentation produced in Quarto?  I know how to add notes to all other slides, but not the **title** slide.  (I don't think it matters, but I'm using JupyterLab.)

**Comment by mcanouil** [comment]
You'll have to modify the title slide partial, see <https://quarto.org/docs/journals/templates.html#revealjs-partials>.
I don't believe there is another way to do that.

Note that, you could also add a yaml key "note" into the partial which would allow you to specify notes for the title slide.

**Comment by cderv** [comment, marked as answer]
Link to revealjs doc about this too: https://quarto.org/docs/presentations/revealjs/advanced.html#custom-template

Otherwise, there is probably an option using a revealjs specificity, and Pandoc feature. 

* From revealjs doc https://revealjs.com/speaker-view/ you can use a `data-notes` attributes
> Alternatively you can add your notes in a data-notes attribute on the slide. Like `<section data-notes="Something important"></section>`.

* Considering how Pandoc, and thus Quarto works, you can add attributes to title slide using `title-slide-attributes`

Try this example 

````markdown
---
title: Test
title-slide-attributes: 
  data-notes: Some speaker notes
format: revealjs
---

## content 

::: notes
Some other notes
:::
````

See also this comment https://github.com/jgm/pandoc/issues/5237#issuecomment-804267358 that gives some details.

Hope it helps

**Reply by jimjam-slam** [reply]
I do a twist on this where I leave the the title out of the frontmatter, and then I use an HTML slide title instead of a Markdown one (so that the h2 isn't used to insert a blank slide). I think essentially I'm just cancelling the title slide altogether.

````md
---
format: revealjs
---

::: {.notes}
Hi, I'm James! I should say some things!
:::

<h2>The best talk ever</h2>

By James

## It has a second slide

...
````

**Reply by cderv** [reply]
Yes, only side effect with this is that you don't get any Title slide styling. The first slide is just a regular slide. 

Not sure about the `<h2>` tag because if you don't use it , you should get the same result

```markdown
---
format: revealjs
---

## The best talk ever

By James

::: {.notes}
Hi, I'm James! I should say some things!
:::

## It has a second slide

...

````

Overall you are indeed removing title slide and using custom normal slide, so that works.

**Reply by jimjam-slam** [reply]
Ahh yeah, I forgot—I'm just styling the first slide myself 😅
